pytissueoptics/scene/geometry/vector.py

```diff
@@ -91,3 +94,48 @@ def update(self, x: float, y: float, z: float):
 
     def copy(self) -> 'Vector':
         return Vector(self._x, self._y, self._z)
+
+    def rotateAround(self, unitAxis: 'Vector', theta: float):
+        # This is the most expensive (and most common)
+        # operation when performing Monte Carlo in tissue
+        # (15% of time spent here). It is difficult to optimize without
+        # making it even less readable than it currently is
+        # http://en.wikipedia.org/wiki/Rotation_matrix
+        #
+        # Several options were tried in the past such as
+        # external not-so-portable C library, unreadable
+        # shortcuts, sine and cosine lookup tables, etc...
+        # and the performance gain was minimal (<20%).
+        # For now, this is the best, most readable solution.
+
+        cost = math.cos(theta)
+        sint = math.sin(theta)
+        one_cost = 1 - cost
+
+        ux = unitAxis.x
+        uy = unitAxis.y
+        uz = unitAxis.z
+
+        X = self.x
+        Y = self.y
+        Z = self.z
+
+        x = (cost + ux * ux * one_cost) * X \
+            + (ux * uy * one_cost - uz * sint) * Y \
+            + (ux * uz * one_cost + uy * sint) * Z
+        y = (uy * ux * one_cost + uz * sint) * X \
+            + (cost + uy * uy * one_cost) * Y \
+            + (uy * uz * one_cost - ux * sint) * Z
+        z = (uz * ux * one_cost - uy * sint) * X \
+            + (uz * uy * one_cost + ux * sint) * Y \
+            + (cost + uz * uz * one_cost) * Z
+
+        self.update(x, y, z)
+
+    def anyPerpendicular(self) -> 'Vector':
```

Comment:
https://math.stackexchange.com/questions/133177/finding-a-unit-vector-perpendicular-to-another-vector
COmparing with this, your answer seems just fine.

Reply:
comparing with which answer ?

Reply:
the algorithm summary, answer number 3

Reply:
On pourrait faire comme lui, mais ça rajouterait un gros overhead à vérifier lequel n'est pas négatif

Reply:
pas trop sure stun overhead si cest dans la fonction, mais bon je veux un algo stable surtout. La rapidité importe peu car on le call juste une fois par photon.